WooCommerce/Classes/ViewRelated/Products/Variations/Bulk Update/BulkUpdateViewController.swift

```diff
@@ -0,0 +1,180 @@
+    func removePlaceholder() {
+        tableView.removeGhostContent()
+    }
+
+    /// Configures the table view: registers Nibs & setup datasource / delegate
+    ///
+    func configureTableView() {
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.backgroundColor = .listBackground
+
+        registerTableViewHeaderSections()
+        registerTableViewCells()
+
+        tableView.dataSource = self
+        tableView.delegate = self
+    }
+
+    func registerTableViewHeaderSections() {
+        let headerNib = UINib(nibName: TwoColumnSectionHeaderView.reuseIdentifier, bundle: nil)
+        tableView.register(headerNib, forHeaderFooterViewReuseIdentifier: TwoColumnSectionHeaderView.reuseIdentifier)
+    }
+
+    func registerTableViewCells() {
+        for row in Row.allCases {
+            tableView.registerNib(for: row.type)
+        }
+    }
+
+    var sections: [Section] {
+        return viewModel.sections
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension BulkUpdateViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return viewModel.sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.sections[section].rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = rowAtIndexPath(indexPath)
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension BulkUpdateViewController: UITableViewDelegate {
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, estimatedHeightForHeaderInSection section: Int) -> CGFloat {
+        return Constants.sectionHeight
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let leftText = viewModel.sections[section].title else {
+            return nil
+        }
+
+        let reuseIdentifier = TwoColumnSectionHeaderView.reuseIdentifier
+        guard let headerView = tableView.dequeueReusableHeaderFooterView(withIdentifier: reuseIdentifier) as? TwoColumnSectionHeaderView else {
+            fatalError("Could not find section header view for reuseIdentifier \(reuseIdentifier)")
+        }
+
+        headerView.leftText = leftText
+        headerView.rightText = nil
+
+        return headerView
+    }
+}
+
+// MARK: - Convenience Methods
+//
+private extension BulkUpdateViewController {
+    func rowAtIndexPath(_ indexPath: IndexPath) -> Row {
+        return sections[indexPath.section].rows[indexPath.row]
+    }
+}
+
+// MARK: - Private Types
+//
+extension BulkUpdateViewController {
```

Comment:
These don't seem as private types, they are used in the VM, right?

Reply:
yes they are to be used in the VM also, since it will provide the array of sections (currently only one), for this PR the VC only needs the type of the `UITableViewCell` to register it(for the GhostCells).

Reply:
That is true, I removed them!

Reply:
That is true, I removed them!